extensions/blocks/contact-form/edit.js

```diff
@@ -239,78 +237,124 @@ class JetpackContactForm extends Component {
 		return fieldEmailError && fieldEmailError.length > 0;
 	}
 
+	setVariation( variation ) {
+		const { setAttributes, replaceInnerBlocks, clientId, selectBlock } = this.props;
+
+		if ( variation.attributes ) {
+			setAttributes( variation.attributes );
+		}
+
+		if ( variation.innerBlocks ) {
+			replaceInnerBlocks(
+				clientId,
+				this.createBlocksFromInnerBlocksTemplate( variation.innerBlocks )
+			);
+		}
+
+		selectBlock( clientId );
+	}
+
+	createBlocksFromInnerBlocksTemplate( innerBlocksTemplate ) {
+		const blocks = map( innerBlocksTemplate, ( [ name, attributes, innerBlocks = [] ] ) =>
+			createBlock( name, attributes, this.createBlocksFromInnerBlocksTemplate( innerBlocks ) )
+		);
+
+		return blocks;
+	}
+
 	render() {
-		const { className, attributes } = this.props;
-		const { hasFormSettingsSet } = attributes;
-		const formClassnames = classnames( className, 'jetpack-contact-form', {
-			'has-intro': ! hasFormSettingsSet,
-		} );
+		const { className, blockType, variations, defaultVariation, hasInnerBlocks } = this.props;
+
+		const formClassnames = classnames( className, 'jetpack-contact-form' );
+
+		if ( ! hasInnerBlocks && registerBlockVariation ) {
+			return (
+				<div className={ formClassnames }>
+					<BlockVariationPicker
+						icon={ get( blockType, [ 'icon', 'src' ] ) }
+						label={ get( blockType, [ 'title' ] ) }
+						instructions={ __(
+							"Please select which type of form you'd like to add, or create your own using the skip option.",
+							'jetpack'
+						) }
+						variations={ variations }
+						allowSkip
+						onSelect={ ( nextVariation = defaultVariation ) => {
+							this.setVariation( nextVariation );
+						} }
+					/>
+				</div>
+			);
+		}
 
 		return (
-			<Fragment>
+			<>
+				{ ToolbarGroup && (
+					<BlockControls>
+						<ToolbarGroup>
+							<Dropdown
+								position="bottom right"
+								className="jetpack-contact-form-settings-selector"
+								contentClassName="jetpack-contact-form__popover"
+								renderToggle={ ( { isOpen, onToggle } ) => {
+									const openOnArrowDown = event => {
+										if ( ! isOpen && event.keyCode === DOWN ) {
+											event.preventDefault();
+											event.stopPropagation();
+											onToggle();
+										}
+									};
+
+									return (
+										<Button
+											className="components-toolbar__control jetpack-contact-form__toggle"
+											label={ __( 'Edit Form Settings' ) }
+											onClick={ onToggle }
+											onKeyDown={ openOnArrowDown }
+											icon={ <Icon icon="edit" /> }
+										/>
+									);
+								} }
```

Comment:
Nitpick: we could make this more readable by moving it into its own render method.

Reply:
Done. 👍